integration_test/mssql/all_test.exs

```diff
@@ -0,0 +1,21 @@
+ecto = Mix.Project.deps_paths()[:ecto]
+Code.require_file "#{ecto}/integration_test/cases/assoc.exs", __DIR__
+Code.require_file "#{ecto}/integration_test/cases/interval.exs", __DIR__
+Code.require_file "#{ecto}/integration_test/cases/joins.exs", __DIR__
+Code.require_file "#{ecto}/integration_test/cases/preload.exs", __DIR__
+Code.require_file "#{ecto}/integration_test/cases/repo.exs", __DIR__
+Code.require_file "#{ecto}/integration_test/cases/type.exs", __DIR__
+
+ Code.require_file "../sql/alter.exs", __DIR__
+ # lock for update works, but other unit tests require transaction snapshot isolation to be ON
+ # so this test is skipped and must run manually
+ # Code.require_file "../sql/lock.exs", __DIR__
```

Comment:
We can enable this now, right? Or is transaction snapshot not working yet?

Reply:
I tried to use change I made in tds but unfortunately no. Reson is that `TestRepo` is **sandboxed** and it begins transaction implicitly in sandbox module. So I can't pass` isolation_level` to `handle_begin` with `mode: transaction` only for mode: **savepoint**. 

I'm thinking to add attribute tag `@tag isolation_level: :snapshot` that should be used per test case, something like we did for DbConnection and `@tag mode: :transaction`. 

There is second option but this could interfere other adapters,  if we use [`Sandbox` :isolation option](https://github.com/mjaric/ecto_sql/blob/d5257f2b9a9054381e82e54b48c95cfd51a1e65b/lib/ecto/adapters/sql/sandbox.ex#L432), I'm not 100% sure that this will work since tds use ["Transaction Manager"](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-tds/0fb28ba5-ddcb-4d02-95c3-aa5b05ec6092), but it is worth exploring as another option

I think it would be best if `Ecto.Integration.Case` implementation reads this `@tag` and use it in [setup](https://github.com/mjaric/ecto_sql/blob/a2fd31073b0910a00bae4f292731b03012d86411/integration_test/mssql/test_helper.exs#L143) so each adapter can decide if this tag should be used or not. Make sense?